algorithm.go

```diff
@@ -48,6 +51,29 @@ const (
 // RFC 8152 16.4: https://datatracker.ietf.org/doc/html/rfc8152#section-16.4
 type Algorithm int64
 
+// AlgorithmFromString returns the Algorithm corresponding to the specified
+// name.
+func AlgorithmFromString(v string) (Algorithm, error) {
```

Comment:
just a stylistic note: returning `error` here seems slightly redundant given we have defined `AlgorithmInvalid`.

Reply:
Yeah, that crossed my mind. I figured it would be more idiomatic to use a separate error, but I'm happy to simplify.